<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="en"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../.resources/report.css" type="text/css"/><link rel="shortcut icon" href="../.resources/report.gif" type="image/gif"/><title>TestFailure.java</title><link rel="stylesheet" href="../.resources/prettify.css" type="text/css"/><script type="text/javascript" src="../.resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../.sessions.html" class="el_session">Sessions</a></span><a href="../index.html" class="el_report">Junit-r4.11</a> &gt; <a href="index.source.html" class="el_package">junit.framework</a> &gt; <span class="el_source">TestFailure.java</span></div><h1>TestFailure.java</h1><pre class="source lang-java linenums">package junit.framework;

import java.io.PrintWriter;
import java.io.StringWriter;


/**
 * A &lt;code&gt;TestFailure&lt;/code&gt; collects a failed test together with
 * the caught exception.
 *
 * @see TestResult
 */
public class TestFailure extends Object {
    protected Test fFailedTest;
    protected Throwable fThrownException;


    /**
     * Constructs a TestFailure with the given test and exception.
     */
<span class="fc" id="L21">    public TestFailure(Test failedTest, Throwable thrownException) {</span>
<span class="fc" id="L22">        fFailedTest = failedTest;</span>
<span class="fc" id="L23">        fThrownException = thrownException;</span>
<span class="fc" id="L24">    }</span>

    /**
     * Gets the failed test.
     */
    public Test failedTest() {
<span class="nc" id="L30">        return fFailedTest;</span>
    }

    /**
     * Gets the thrown exception.
     */
    public Throwable thrownException() {
<span class="fc" id="L37">        return fThrownException;</span>
    }

    /**
     * Returns a short description of the failure.
     */
    @Override
    public String toString() {
<span class="nc" id="L45">        StringBuffer buffer = new StringBuffer();</span>
<span class="nc" id="L46">        buffer.append(fFailedTest + &quot;: &quot; + fThrownException.getMessage());</span>
<span class="nc" id="L47">        return buffer.toString();</span>
    }

    public String trace() {
<span class="nc" id="L51">        StringWriter stringWriter = new StringWriter();</span>
<span class="nc" id="L52">        PrintWriter writer = new PrintWriter(stringWriter);</span>
<span class="nc" id="L53">        thrownException().printStackTrace(writer);</span>
<span class="nc" id="L54">        StringBuffer buffer = stringWriter.getBuffer();</span>
<span class="nc" id="L55">        return buffer.toString();</span>
    }

    public String exceptionMessage() {
<span class="fc" id="L59">        return thrownException().getMessage();</span>
    }

    public boolean isFailure() {
<span class="nc" id="L63">        return thrownException() instanceof AssertionFailedError;</span>
    }
}
</pre><div class="footer"><span class="right">Created with <a href="http://www.eclemma.org/jacoco">JaCoCo</a> 0.7.0.201403182114</span></div></body></html>